can/io/generic.py

```diff
@@ -3,19 +3,20 @@
 """
 
 from abc import ABCMeta
-from typing import Any, Optional, cast, Union, TextIO, BinaryIO, Type
+from typing import Any, Optional, cast, Iterable, Union, TextIO, BinaryIO, Type, ContextManager
+from typing_extensions import Literal
```

Comment:
We do not have the `typing_extensions` dependency in setup.py. Do we need it?

Reply:
Yeah, also realized that. It somehow was present in my venv though so I thought I'd give it a try.

I'd propose to add it, yes. It's the backport of newer typing additions to older Python versions (the `Literal` was [added in 3.8](https://docs.python.org/3/library/typing.html#typing.Literal)). Shall I add it to this PR?

Reply:
Personally i don't mind it.